install.sh

```diff
@@ -0,0 +1,151 @@
+#! /bin/bash
+
+### Checking for prerequisites ###
+
+#check if running as root
+if [[ $EUID -ne 0 ]]; then
+   echo "This script must be run as root" 
+   exit 1
+fi
+
+#check for docker
+if ! command -v docker &> /dev/null
+then
+  echo "Docker is not installed on this host. Please install and run this script again."
+  exit 1
+fi
+
+#check for docker-compose and install if not currently
+if command -v docker-compose &> /dev/null
+then
+  echo "docker-compose is installed install will continue"
+else
+  read -p "docker-compose not detected as installed. Would you like to install it now? (y/n): " -n 1 -r 
+  echo "/n"
+  if [[ $REPLY =~ ^[yY]$ ]]
+  then
+    curl -L "https://github.com/docker/compose/releases/download/1.29.2/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
+    chmod +x /usr/local/bin/docker-compose
+  fi
+fi
+
+### Checking commandline strings ###
+
+if [[ $* == *--[sS][sS][Ll]* ]]
+then
+    echo "Installing in HTTPS mode"
+    SSL="yes"
+else 
+    echo "installing in HTTP mode"
+    SSL="no"
+fi
+
+mkdir ./nginx ./gitlab
+
+if [ $SSL == "yes" ]
+then
+    #Make folder for SSL certs which will be mounted at /tmp on the container
+    mkdir ./nginx/ssl
+    #Read in the NGINX SSL config file 
+    #certs will need to be placed in the ./nginx/ssl/ folder before the container is started
+    echo "server {
+  listen 443 SSL;
+  ssl_certificate /tmp/nginx.crt;
+  ssl_certificate_key /tmp/nginx.key;
+  location / {
+    proxy_pass http://gitlabsvr/;
+  }
+}"> ./nginx/nginx.conf
```

Comment:
I don't usually see echo used for multi-line strings. Obviously, you've made it work for you so pick what you like better. I usually use something called a heredoc, which would look like this. The `EOF` is an arbitrary string you choose to delimit the end of your text. It just has to be on a line by itself at the end.

```bash
    cat <<EOF > ./nginx/nginx.conf
server {
  listen 443 SSL;
  ssl_certificate /tmp/nginx.crt;
  ssl_certificate_key /tmp/nginx.key;
  location / {
    proxy_pass http://gitlabsvr/;
  }
}
EOF
```

The advantages of a heredoc is that you can use any quote inside without worrying about escaping or terminating your string too early.

Heredocs will interpret bash variables starting with a dollar sign e.g. `$somevar`. If your text has dollar signs that you don't want interpreted that way you can either escape them all individually in the text like `\$somevar` or just escape the delimiter at the beginning like this, which disables interpreting all variables in the text.

```bash
    cat <<\EOF > ./nginx/nginx.conf
server {
  listen 443 SSL;
  ssl_certificate /tmp/nginx.crt;
  ssl_certificate_key /tmp/nginx.key;
  location / {
    proxy_pass http://gitlabsvr/;
  }
}
EOF
```